engine/player/unique_gear_dragonflight.cpp

```diff
@@ -7113,6 +7113,167 @@ void gift_of_ursine_vengeance( special_effect_t& effect )
   }
 }
 
+// Fyrakk's Tainted Rageheart
+
+// 422652 Tainted Heart
+// Effect 1 Self / Enemy Tick Passive
+// Effect 2 On-Use Self Tick
+// Effect 3 On-Use Absorb
+// Effect 4 On-Use Enemy Tick
+
+// 425461 Self Damage
+// 425461 Enemy Damage
+// 422750 On-Use Debuff
+// 425571 On-Use Absorb
+// 425703 Self Damage (On-Use)
+// 425701 Enemy Damage (On-Use)
+void fyrakks_tainted_rageheart( special_effect_t& effect )
+{
+  struct tainted_heart_t : public proc_spell_t
+  {
+    struct self_damage_t : public generic_aoe_proc_t
+    {
+      self_damage_t( const special_effect_t& e )
+        : generic_aoe_proc_t( e, "tainted_heart_self_damage", e.player->find_spell( 425461 ) )
+      {
+        background  = true;
+        aoe = 1;
+        target      = e.player;
+        stats->type = stats_e::STATS_NEUTRAL;
+        callbacks   = false;  // TODO: confirm if this triggers any proc flags.
+        base_dd_min = base_dd_max = e.player->find_spell( 422652 )->effectN( 1 ).average( e.item );
+      }
+    };
+
+    struct enemy_damage_t : public generic_aoe_proc_t
+    {
+      enemy_damage_t( const special_effect_t& e )
+        : generic_aoe_proc_t( e, "tainted_heart_enemy_damage", e.player->find_spell( 425461 ) )
+      {
+        // TODO: Add split/aoe behaviour.
+        aoe              = 8;
+        split_aoe_damage = false;
+        aoe_damage_increase = false;
+        background       = true;
+        base_dd_min = base_dd_max = e.player->find_spell( 422652 )->effectN( 1 ).average( e.item );
+      }
+    };
+
+    self_damage_t* self_damage;
+    enemy_damage_t* enemy_damage;
+
+    tainted_heart_t( const special_effect_t& e ) : proc_spell_t( "tainted_heart", e.player, e.driver(), e.item )
+    {
+      self_damage  = new self_damage_t( e );
+      enemy_damage = new enemy_damage_t( e );
+
+      add_child( self_damage );
+      add_child( enemy_damage );
+    }
+
+    void execute() override
+    {
+      self_damage->execute();
+      enemy_damage->execute();
+      proc_spell_t::execute();
+    }
+  };
+
+  struct shadowflame_rage_t : public proc_spell_t
+  {
+    struct self_damage_t : public generic_aoe_proc_t
+    {
+      self_damage_t( const special_effect_t& e )
+        : generic_aoe_proc_t( e, "shadowflame_rage_self_damage", e.player->find_spell( 425703 ) )
+      {
+        background  = true;
+        aoe = 0;
+        split_aoe_damage = false;
+        aoe_damage_increase = false;
+        target      = e.player;
+        stats->type = stats_e::STATS_NEUTRAL;
+        callbacks   = false;  // TODO: confirm if this triggers any proc flags.
+        base_dd_min = base_dd_max = e.player->find_spell( 422652 )->effectN( 2 ).average( e.item );
+      }
+    };
+
+    struct enemy_damage_t : public generic_aoe_proc_t
+    {
+      enemy_damage_t( const special_effect_t& e )
+        : generic_aoe_proc_t( e, "shadowflame_lash_enemy", e.player->find_spell( 425701 ) )
```

Comment:
```suggestion
        : generic_aoe_proc_t( e, "shadowflame_lash_enemy", e.player->find_spell( 425701 ), true )
```

setting the bool true here will handle the 15% increase per target up to 5 that trinkets often do, with all its behavior pre-set.